compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -3500,25 +3500,29 @@ private void validateAnnotationAttachmentExpr(BLangAnnotationAttachment annAttac
             return;
         }
 
-
         // At this point the type is a subtype of  map<anydata>|record{ anydata...; } or
-        // map<anydata>[]|record{ anydata...; }[], thus an expression is required.
-        if (annAttachmentNode.expr == null) {
+        // map<anydata>[]|record{ anydata...; }[]. For map<anydata>[]|record{ anydata...; }[] an expression is
+        // required while map<anydata>|record{ anydata...; } can be empty.
+        if (annAttachmentNode.expr == null && annotationSymbol.attachedType != null &&
+                !(annotationSymbol.attachedType.type.tag == TypeTags.MAP ||
```

Comment:
When annotation declaration support multiple annotations, `annotationSymbol.attachedType` is a list, IMO we don't need to provide empty mapping constructor for these scenarios.

```ballerina
type T map<int>;
annotation T[] ANNOT on function;

@ANNOT {}
@ANNOT {}
public function main(string... argv) {
}
```

Reply:
Also current changes allow below code, which is wrong as we can't provide empty mapping constructor here.
```ballerina
type T record { int i; };

annotation T ANNOT on function;

@ANNOT
public function main(string... argv) {
}
```

Reply:
My comments are as same as @MaryamZi please ignore these.